File: Algorithms/Math/angleClock.cpp
class Solution {
public:
    double angleClock(int hours, int minutes) {
        /* 60 minutes/ticks * 60 - gives degree*/
        double minutesDegree = minutes * 6;
        double hoursDegree = (hours * 5)*6;
        /* Max degree must be less then 30 degrees */
        hoursDegree += static_cast<double>(minutes)/2.0;
        double ans = (minutesDegree - hoursDegree);

        while(ans < 0) ans += 360;
        while(ans >= 360) ans -= 360;

        if(ans > 180) return 360.0 - ans;
        return ans; 
    }
};

File: Algorithms/Math/angleClock.py
class Solution:
    def angleClock(self, hour: int, minutes: int) -> float:

        minutesDegree = (minutes * 6) % 360
        hoursDegree = (hour) * 30 
        # there 5 ticks between the each hour - each represent 6 degree - which gives us 30
        hoursDegree += (float(minutes)/60.0) * 30

        ans = hoursDegree - minutesDegree

        while ans > 360: ans -= 360
        while ans < 0: ans += 360

        if ans > 180: ans = 360 - ans

        return ans